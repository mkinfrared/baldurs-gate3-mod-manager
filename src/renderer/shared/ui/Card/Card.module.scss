.Card {
  @layer base {
    --background: var(--surface-2);
    --border-color: var(--primary-3);

    @media (prefers-color-scheme: light) {
      --background: var(--surface-1);
    }

    background-color: var(--background);
    border: 1px solid var(--border-color);
    padding: 1rem;
  }
}
