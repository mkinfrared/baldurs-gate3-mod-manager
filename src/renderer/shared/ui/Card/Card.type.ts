import { ReactNode } from "react";

export type CardProps = {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
  children?: ReactNode;
};
