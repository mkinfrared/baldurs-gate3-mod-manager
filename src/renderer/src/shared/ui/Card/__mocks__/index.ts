export { default } from "./Card.mock";
