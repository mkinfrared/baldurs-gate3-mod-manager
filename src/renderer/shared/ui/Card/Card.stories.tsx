import { Meta, StoryObj } from "@storybook/react";

import Card from "./Card";
import css from "./Story.module.scss";

type Story = StoryObj<typeof meta>;

const meta = {
  title: "UI/Card",
  component: Card,
  tags: ["autodocs"],
} satisfies Meta<typeof Card>;

const Default: Story = {
  args: {},
  render: (args) => (
    <div className={css.Story}>
      <Card {...args} />
    </div>
  ),
};

export { Default };

export default meta;
