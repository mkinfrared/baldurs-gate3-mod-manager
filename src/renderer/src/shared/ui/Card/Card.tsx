import { classNames } from "@renderer/shared/lib/helpers";

import css from "./Card.module.scss";
import { CardProps } from "./Card.type";

const Card = ({ className, children }: CardProps) => (
  <div className={classNames(css.Card, className)} data-testid="Card">
    {children}
  </div>
);

export default Card;
