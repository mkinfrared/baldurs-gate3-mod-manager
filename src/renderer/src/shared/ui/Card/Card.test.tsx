import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import Card from "./Card";

describe("<Card />", () => {
  const Component = <Card />;

  it("should be defined", () => {
    expect(Card).toBeDefined();
  });

  it("should match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });

  it("should contain a data test id", () => {
    const { getByTestId } = render(Component);
    const element = getByTestId("Card");

    expect(element).toBeDefined();
  });
});
