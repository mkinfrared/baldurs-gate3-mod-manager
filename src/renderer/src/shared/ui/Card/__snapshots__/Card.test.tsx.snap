// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Card /> > should match the snapshot 1`] = `
<div>
  <div
    class="Card"
    data-testid="Card"
  />
</div>
`;
